hw09-custom-orm/src/main/java/otus/jdbc/mapper/impl/EntityClassMetaDataImpl.java

```diff
@@ -0,0 +1,83 @@
+package otus.jdbc.mapper.impl;
+
+import otus.jdbc.mapper.core.metadata.EntityClassMetaData;
+import otus.jdbc.mapper.core.metadata.Id;
+import otus.jdbc.mapper.core.exceptions.MetaDataExtractionException;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+
+import static otus.jdbc.mapper.core.exceptions.MetaDataExtractionException.NOT_CONSTRUCTOR_WITH_NO_ARGUMENTS;
+import static otus.jdbc.mapper.core.exceptions.MetaDataExtractionException.NO_ID_ANNOTATION;
+
+public class EntityClassMetaDataImpl<T> implements EntityClassMetaData<T> {
+
+    private final String name;
+    private final Constructor<T> constructor;
+    private final Field idField;
+    private final List<Field> allFields;
+    private final List<Field> fieldsWithoutId;
+
+    public EntityClassMetaDataImpl(Class<T> entityClass) {
+        this.name = entityClass.getSimpleName();
```

Comment:
Лучше сам класс сохранить, а не просто имя в виде строки.
```suggestion
        this.clazz = entityClass;
```